Sources/CvPlayer.cpp

```diff
@@ -16687,13 +16687,12 @@ void CvPlayer::setEndTurn(bool bNewValue)
 /*                                                                                              */
 /* Advanced Automations                                                                         */
 /************************************************************************************************/
-			int iLoop;
-			for(CvSelectionGroup* pLoopSelectionGroup = firstSelectionGroup(&iLoop); pLoopSelectionGroup; pLoopSelectionGroup = nextSelectionGroup(&iLoop))
+			foreach_ (CvSelectionGroup* group, groups())
```

Comment:
This is fine how it is, note for future reference you can do this though:
```cpp
foreach_(CvSelectionGroup* group, groups() | filtered(CvSelectionGroup::fn::getAutomateType() == AUTOMATE_SHADOW))
{
	group->setForceUpdate(true);
	group->AI_update();
}
```
(if you implement the fn in the same manner as is done for the CvUnit ones)

Reply:
will do.